src/hooks/useMountedState.ts

```diff
@@ -31,16 +31,19 @@ import { useIsMounted } from './useIsMounted';
  * D is the type passed when then useMountedState is used
  * e.g. const [pass, setPass] = useMountedState<string>('someString');
  */
-export const useMountedState = <D>(initialValue: D) => {
+export const useMountedState = <D>(initialState: D | (() => D)) => {
   const isMounted = useIsMounted();
 
-  const [state, setState] = useState<D>(initialValue);
+  const [state, setState] = useState<D>(initialState);
 
-  const setMountedState = (value: D) => {
-    if (isMounted()) {
-      setState(value);
-    }
-  };
+  const setMountedState = useCallback(
+    (value: D) => {
+      if (isMounted()) {
+        setState(value);
+      }
+    },
+    [isMounted]
```

Comment:
Should we have `setState` here? I guess it doesn't change anyway.

Reply:
Im not sure I understand. setState is the only actuon that exists inside the callback (and the reason why we created the useMountedState). if we remove that - then.... what remains?

Reply:
I meant to add `setState` to the array of dependencies, like so:
```
[isMounted, setState]
```
so that in case `setState` function changes the callback is created anew.

Reply:
if added - `setState` is just a dependency - and as assigned as useState at the beginning of the component Im not sure that it would make any difference, to be added where you propose.